clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2375,24 +2376,30 @@ void handleResponse(AbstractResponse abstractResponse) {
                                 final String protocolType = groupMetadata.protocolType();
                                 if (protocolType.equals(ConsumerProtocol.PROTOCOL_TYPE) || protocolType.isEmpty()) {
                                     final List<DescribeGroupsResponse.GroupMember> members = groupMetadata.members();
-                                    final List<MemberDescription> consumers = new ArrayList<>(members.size());
+                                    final List<MemberDescription> memberDescriptions = new ArrayList<>(members.size());
 
                                     for (DescribeGroupsResponse.GroupMember groupMember : members) {
-                                        final PartitionAssignor.Assignment assignment =
+                                        if (groupMember.memberAssignment().remaining() > 0) {
```

Comment:
@cmccabe Could you help me understand when would member assignment buffer be empty? The patch excludes a member from the group description if the member assignment if empty. In that case, would it be more reasonable to make change in the server side such that the member is not included in the DescribeGroupsResponse in the first place?

Reply:
I agree that it's odd that the server sends empty assignments, and probably should be changed.  However, the 1.2 AdminClient may be used with a 1.1 or older broker, so we still have to handle the old (arguably broken) behavior.  I think we can do a little better here and keep the rest of the member description even if the member assignment is empty.

Reply:
I see. If 1.1 broker will send empty assignment in a for a given member in a legitimate DescribeGroupsResponse, is other information of the `MemberDescription` (e.g. memberId, host) still useful? I am wondering if we should still include the member in the group description instead of discarding it entirely.

Another question is that, if it is not reasonable to return an empty assignment, can we also fix it behavior in the server? If it is not straightforward then we can do it a separate patch.

Reply:
We will return an empty assignment while the group is rebalancing. Rebalances can sometimes take a little while to complete, and we have found it useful to still be able to show the members when the group is in this state.

Reply:
@hachikuji Ah I see. Thanks for the explanation. The logic seems more reasonable now after we include the member in the ConsumerGroupDescription even if the member's assignment is empty.